Zend/zend_API.h

```diff
@@ -728,6 +730,36 @@ ZEND_API void zend_fcall_info_argn(zend_fcall_info *fci, uint32_t argc, ...);
  */
 ZEND_API zend_result zend_fcall_info_call(zend_fcall_info *fci, zend_fcall_info_cache *fcc, zval *retval, zval *args);
 
+/* Zend FCC API to store and handle PHP userland functions */
+static zend_always_inline bool zend_fcc_equals(const zend_fcall_info_cache* a, const zend_fcall_info_cache* b)
+{
+	return a->function_handler == b->function_handler
+		&& a->object == b->object
+		&& a->calling_scope == b->calling_scope
+		&& a->closure == b->closure
+	;
+}
+static zend_always_inline void zend_fcc_addref(zend_fcall_info_cache *fcc)
+{
+	if (fcc->object) {
+		GC_ADDREF(fcc->object);
+	}
+	if (fcc->closure) {
+		GC_ADDREF(fcc->closure);
+	}
+}
+static zend_always_inline void zend_fcc_delref(zend_fcall_info_cache *fcc)
```

Comment:
I would suggest the name dtor instead of delref here, as usually the delref functions only decrement the reference count

Reply:
Indeed, makes more sense. :)